providers/ms_entra_id.go

```diff
@@ -89,6 +95,65 @@ func (p *MicrosoftEntraIDProvider) ValidateSession(ctx context.Context, session
 	return p.OIDCProvider.ValidateSession(ctx, session)
 }
 
+// Redeem exchanges the OAuth2 authentication token for an ID token, considering federated token authentication
+func (p *MicrosoftEntraIDProvider) Redeem(ctx context.Context, redirectURL, code, codeVerifier string) (*sessions.SessionState, error) {
+
+	// Perform usual Redeem for client secret authentication
+	if !p.federatedTokenAuthEnabled {
+		return p.OIDCProvider.Redeem(ctx, redirectURL, code, codeVerifier)
+	}
+
+	return p.redeemWithFederatedToken(ctx, redirectURL, code, codeVerifier)
+}
```

Comment:
Shouldn't we reorder this to make it slightly more readable. I don't like to use negate if it isn't necessary

```suggestion
func (p *MicrosoftEntraIDProvider) Redeem(ctx context.Context, redirectURL, code, codeVerifier string) (*sessions.SessionState, error) {
	if p.federatedTokenAuthEnabled {
		return p.redeemWithFederatedToken(ctx, redirectURL, code, codeVerifier)
	}
	
	return p.OIDCProvider.Redeem(ctx, redirectURL, code, codeVerifier)
}
```